<?xml version="1.0"?>
<robot xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
       xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
       xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
       xmlns:xacro="http://ros.org/wiki/xacro"
>
  <!-- include common properties -->
  <xacro:include filename="$(find getit_robot_gazebo)/urdf/chaser/common.xacro" />

  <!-- topic properties -->
  <xacro:property name="chaser_cog_topic" value="/Robot_1/pose"/>

  <!-- base properties -->
  <xacro:property name="base_x_length" value="3.0"/>
  <xacro:property name="base_y_length" value="0.4998"/>
  <xacro:property name="base_z_length" value="1.0"/>
  <xacro:property name="base_mass" value="2000.0"/>
  <xacro:property name="base_inertia_lx" value="1400.0"/>
  <xacro:property name="base_inertia_ly" value="1400.0"/>
  <xacro:property name="base_inertia_lz" value="2040.0"/>

  <!-- arm link1 properties -->
  <xacro:property name="arm_link1_length" value="0.25"/>
  <xacro:property name="arm_link1_radius" value="0.07"/>
  <xacro:property name="arm_link1_mass" value="20.0"/>
  <xacro:property name="arm_link1_inertia_lx" value="0.10"/>
  <xacro:property name="arm_link1_inertia_ly" value="0.10"/>
  <xacro:property name="arm_link1_inertia_lz" value="0.10"/>

  <!-- arm link2 properties -->
  <xacro:property name="arm_link2_length" value="2.5"/>
  <xacro:property name="arm_link2_radius" value="0.07"/>
  <xacro:property name="arm_link2_mass" value="50.0"/>
  <xacro:property name="arm_link2_inertia_lx" value="26.0"/>
  <xacro:property name="arm_link2_inertia_ly" value="26.0"/>
  <xacro:property name="arm_link2_inertia_lz" value="0.25"/>

  <!-- arm link3 properties -->
  <xacro:property name="arm_link3_length" value="2.5"/>
  <xacro:property name="arm_link3_radius" value="0.07"/>
  <xacro:property name="arm_link3_mass" value="50.0"/>
  <xacro:property name="arm_link3_inertia_lx" value="26.0"/>
  <xacro:property name="arm_link3_inertia_ly" value="26.0"/>
  <xacro:property name="arm_link3_inertia_lz" value="0.25"/>

  <!-- arm joint properties -->
  <xacro:property name="arm_joint_link_length" value="0.15"/>
  <xacro:property name="arm_joint_link_radius" value="0.07"/>

  <!-- arm initial position -->
  <xacro:property name="arm_joint_1_position" value="0"/>
  <xacro:property name="arm_joint_2_position" value="${PI*3.0/4.0}"/>
  <xacro:property name="arm_joint_3_position" value="${PI/2.0}"/>
  <!-- <xacro:property name="arm_joint_2_position" value="0"/>
  <xacro:property name="arm_joint_3_position" value="0"/> -->

  <!-- arm limit position -->
  <xacro:property name="arm_joint2_limit_lower" value="${-PI*3.0/4.0}"/>
  <xacro:property name="arm_joint2_limit_upper" value="${PI*3.0/4.0}"/>
  <xacro:property name="arm_joint3_limit_lower" value="${-PI/2.0}"/>
  <xacro:property name="arm_joint3_limit_upper" value="${PI/2.0}"/>
  <xacro:property name="arm_joint_limit_effort" value="100"/>
  <xacro:property name="arm_joint_limit_velocity" value="100"/>

</robot>
